### PR TITLE
OpenStack Object Storage Service (SDK V2)

### PR DESCRIPTION
**Pull request by sharwell**
This pull request adds support for the basic functionality of the OpenStack Object Storage Service V1 to the V2 SDK.

### CODE REVIEW

**Comment by sharwell** [comment]
This is currently blocked on #450 and #452.

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/StorageMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Text;
+    using OpenStack.Collections;
+
+    /// <summary>
+    /// This is the base class for representing metadata associated with resources in the Object Storage Service.
+    /// </summary>
+    /// <threadsafety static="true" instance="false"/>
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/StorageMetadata.cs]: The `StorageMetadata` types (this and derived) are immutable. Both static and instance members are thread-safe.

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/AccountMetadataExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    /// <preliminary/>
+    public static class AccountMetadataExtensions
+    {
+        /// <summary>
+        /// Gets the name of the <c>X-Account-Bytes-Used</c> HTTP header, which the Object Storage Service may
+        /// include with the <see cref="AccountMetadata"/> to specify the total size of all objects stored in the
+        /// Object Storage Service for the account.
+        /// </summary>
+        /// <seealso cref="GetBytesUsed(AccountMetadata)"/>
+        public const string AccountBytesUsed = "X-Account-Bytes-Used";
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/AccountMetadataExtensions.cs]: These fields should probably be `static readonly` unless there is a particular requirement to have them `const` (e.g. using them as an argument to an attribute).

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ExtractArchiveExtension.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
+// Intentionally placed in the outer scope so IProgress<T> resolves to System.IProgress<T> instead of
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ExtractArchiveExtension.cs]: This workaround is no longer required (see a0b77be2cf1a3d5566a40f7af4509824a37db5a2).

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ExtractArchiveExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,194 @@
+﻿// Intentionally placed in the outer scope so IProgress<T> resolves to System.IProgress<T> instead of
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ExtractArchiveExtensions.cs]: This workaround is no longer required (see a0b77be2cf1a3d5566a40f7af4509824a37db5a2).

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ObjectStorageClient.cs [hunk 5]
```diff
@@ -0,0 +1,682 @@
+﻿// Intentionally placed in the outer scope so IProgress<T> resolves to System.IProgress<T> instead of
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ObjectStorageClient.cs]: This workaround is no longer required (see a0b77be2cf1a3d5566a40f7af4509824a37db5a2).

File: src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ObjectStorageServiceExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,842 @@
+﻿// Intentionally placed in the outer scope so IProgress<T> resolves to System.IProgress<T> instead of
```
Comment (sharwell) [inline src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/ObjectStorageServiceExtensions.cs]: This workaround is no longer required (see a0b77be2cf1a3d5566a40f7af4509824a37db5a2).

**Comment by carolynvs** [comment]
The target branch is invalid now #510 has split v1 and v2 into separate branches.

**Pull request closed without merging**